### DISCUSSION TITLE
Why can't the user agent be changed?

### DISCUSSION THREAD

**Discussion by josesa** [ktr0731/evans#494] [Q&A]
In the code, the `AddHeader` function explicitly checks for the `User-Agent` entry and ignores it.

I could not find in the documentation (or code) any reasoning behind this.

**Comment by ktr0731** [comment]
There is no specific reason.
However, due to an implementation problem with the Go library, the user-agent needs to be specified with a special option, unlike other headers. So, the current implementation doesn't allow specifying the user-agent, but it's possible to support the user-agent modification.
